lib/oauth.js

```diff
@@ -558,7 +558,7 @@ internals.encode = function (string) {
 internals.parse = function (payload) {
 
     payload = Buffer.isBuffer(payload) ? payload.toString() : payload;
-    if (payload[0] === '{') {
+    if (/^\s*{/.test(payload)) {
```

Comment:
Considering trim() is nondestructive to the original reference and JSON.parse() ignores leading/trailing whitespaces, then I don't see the possible clobbering.  Why wouldn't something like this work?

```
payload = Buffer.isBuffer(payload) ? payload.toString() : payload;
if (payload.trim()[0] === '{') {
```